src/mito2/src/worker/handle_alter.rs

```diff
@@ -52,8 +54,14 @@ impl<S> RegionWorkerLoop<S> {
                 "Ignores alter request, region id:{}, region schema version {} is greater than request schema version {}",
                 region_id, version.metadata.schema_version, request.schema_version
             );
-            // Returns if it altered.
-            sender.send(Ok(0));
+            // Returns an error.
+            sender.send(
+                InvalidRegionRequestSchemaVersionSnafu {
```

Comment:
We can return this error if `version.metadata.schema_version != request.schema_version`